src/zcash/JoinSplit.cpp

```diff
@@ -266,8 +267,14 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
 
         std::stringstream ss;
         ss << proof;
+        std::string serialized_proof = ss.str();
 
-        return ss.str();
+        boost::array<unsigned char, ZKSNARK_PROOF_SIZE> result_proof;
+        //std::cout << "proof size in bytes when serialized: " << serialized_proof.size() << std::endl;
+        assert(serialized_proof.size() == ZKSNARK_PROOF_SIZE);
+        memcpy(&result_proof[0], &serialized_proof[0], ZKSNARK_PROOF_SIZE);
```

Comment:
I don't think this is correct: `serialized_proof` is a string, and strings don't have to store their contents contiguously in memory. I think you need to call `serialized_proof.c_str()` but I'm not sure how that deals with null characters.